ludwig/schema/features/augmentation/utils.py

```diff
@@ -0,0 +1,157 @@
+from dataclasses import field
+
+from marshmallow import fields, ValidationError
+from marshmallow_dataclass import dataclass
+
+from ludwig.api_annotations import DeveloperAPI
+from ludwig.constants import TYPE
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.features.augmentation.base import BaseAugmentationContainerConfig
+from ludwig.utils.registry import Registry
+
+
+@DeveloperAPI
+@dataclass(repr=False)
+class ImageAugmentationContainerConfig(BaseAugmentationContainerConfig):
+    """Augmentation container for image features."""
+
+    pass
+
+
+# TODO: Is all of this needed?
+# augmentation_registry = Registry()
+#
+#
+# @DeveloperAPI
+# def register_augmentation(name: str):
+#     def wrap(augmentation_config: BaseAugmentationConfig):
+#         augmentation_registry[name] = augmentation_config
+#         return augmentation_config
+#
+#     return wrap
+
+_augmentation_config_registry = Registry()
+
+
+@DeveloperAPI
+def get_augmentation_config_registry() -> Registry:
+    return _augmentation_config_registry
+
+
+def register_augmentation_config(name: str):
+    def wrap(cls):
+        get_augmentation_config_registry()[name] = cls
+        return cls
+
+    return wrap
+
+
+def get_augmentation_cls(name: str):
+    return get_augmentation_config_registry()[name]
+
+
+@DeveloperAPI
+def AugmentationContainerDataclassField(default=None, description=""):
+    """Custom dataclass field that when used inside a dataclass will allow the user to specify an augmentation
+    config.
+
+    Args:
+        default: The default augmentation config to use.
+        description: The description of the augmentation config.
+
+    Returns: Initialized dataclass field that converts an untyped dict with params to an augmentation config.
+    """
+
+    class AugmentationContainerMarshmallowField(fields.Field):
+        """Custom marshmallow field that deserializes a list for a valid augmentation config from the
+        augmentation_registry and creates a corresponding JSON schema for external usage."""
+
+        def _deserialize(self, value, attr, data, **kwargs):
+            if value is None:
```

Comment:
Let's get rid of this line. Fundamentally what it allows is for the user to manually set `augmentation: None` in their Ludwig config. I don't think this should be allowed - either the user should not set `augmentation` in their config at all or they need to provide a list.

Let's instead replace this with an assert `isinstance(value, list)` or some kind of equivalent type checking.

Reply:
done